### ISSUE TITLE
Settings: simplify language picker component

### ISSUE THREAD

**Issue by rickybanister** [Automattic/wp-calypso#41938]
The language picker in the general settings screen is a custom dropdown:

<img width="615" alt="Screen Shot 2020-05-07 at 9 53 35 AM" src="https://user-images.githubusercontent.com/437258/81303389-a1377500-9049-11ea-9999-f5fe7546f946.png">

It has the word 'CHANGE' inside the box indicating it can be clicked, but this doesn't seem necessary. We can make it look exactly like the timezone picker just below.

Something like this:

![image](https://user-images.githubusercontent.com/437258/81303954-65e97600-904a-11ea-8772-29ba7d1e2426.png)

**Comment by davemart-in** [comment]
If we remove the word "CHANGE", I assume we will still need to use a custom dropdown component here @rickybanister.  To clarify, the focus for this issue is around visually consistent not component consistency, right?

**Comment by rickybanister** [comment]
I hacked the inspector to create my mockup above, we likely do not need a custom select menu. We could try using a standard select menu with just the language name in text. 

Or `EN - English` if the code and language name are helpful. 

We should check on accessibility best practices as well though and make sure we can properly render all language names natively in the picker, that kind of thing.

**Comment by davemart-in** [comment]
Alright, if we're talking about replacing the component, I'll move this to the backlog, as I don't think it's super high priority.

**Comment by rickybanister** [comment]
I do think my initial proposal is lighter weight—to just remove the word 'change' from the dropdown and make the height and font-weight align. 

It's not very important, I just don't really understand why this element is so different to begin with :)

**Comment by davemart-in** [comment]
I was chatting with Gary about this and he brought up that this language UI has a completely different UI once you click in which has a search Ui baked in.

**Comment by pento** [comment]
As Dave mentioned, the popup for this provides a better UI than a standard dropdown, which includes search.

![](https://user-images.githubusercontent.com/352291/83471819-06577c80-a4c9-11ea-809e-314078252b9b.png)

Interestingly, the search functionality is quite comprehensive. I can search for languages by their English name, their localised name (including non-latin character sets), and their name in my UI language.

Given how unwieldy it would be if we replaced this with a standard dropdown (or a `SelectDropdown`, which provides largely the same functionality), I'm inclined to think it needs a more comprehensive solution.

Site timezones have worked around this issue by suggesting an option, we could do the same based on the browser's language settings.

![](https://user-images.githubusercontent.com/352291/83472252-22a7e900-a4ca-11ea-851a-de7699e72acc.png)

Alternatively, it appears that Gutenberg is close to merging [exactly the UI component](https://github.com/WordPress/gutenberg/pull/19657) we could use here. Of course, this runs into the problem of how we manage ad-hoc mixing of WordPress components into existing Calypso UIs (vs examples like Gutenboarding, where the UI was created entirely with WordPress components), but I suspect that's a simpler problem to solve than creating a new searchable dropdown component.

**Comment by rickybanister** [comment]
Thanks for looking into this. The real impetus of this issue was the visual appearance of the select menu itself. We can probably do about 10 minutes of css work to make it the same size, height of a regular select and remove the 'change' text. We can keep the popup as is for now.

Longer term I think we are likely to see much more adoption of gutenberg primitives and components, to the point where we consume all of gutenberg within calypso. We would only maintain our own components for elements that serve no purpose in core (like an upgrade nudge or something). We don't need to move in that direction immediately, but whenever practical we should consider doing so. Maintaining less code over time will benefit engineering, and a more cohesive experience across wp-admin, the editor, and wpcom will benefit customers very directly. It's worth stating explicitly that there is no 'brand value' in having different UI components. We can apply a sense of brand through color and typography in a more sustainable way than through proprietary interaction patterns.

**Comment by ncardeli** [comment]
I can take a look at this, following @rickybanister proposal to remove the "change" text and make it look like a regular select.

**Comment by ncardeli** [comment]
I took a closer look at this issue and have a couple of questions:

1. Should the language picker match the height of the timezone select (40px)?

If so, I think it would look too cramped, specially when the selected language has both a language code and a subcode (for example, ES MX - Español de México).

**48px height**
![imagen](https://user-images.githubusercontent.com/5444806/88009594-4f9b7400-cae9-11ea-90b8-162426eb06e4.png)

vs

**40px height**
![imagen](https://user-images.githubusercontent.com/5444806/88009629-604bea00-cae9-11ea-8bdf-0762b13b8ada.png)

2. Should the language picker and the select match the font size, weight and color? I think they should

**Matching font size, weight and color**
![imagen](https://user-images.githubusercontent.com/5444806/88009981-37782480-caea-11ea-9670-5ece5a873bbd.png)

vs

**Non-matching**
![imagen](https://user-images.githubusercontent.com/5444806/88010018-47900400-caea-11ea-9771-814af560bab3.png)

To sum up, this is how I feel it would look best (48px height and matching font):

![imagen](https://user-images.githubusercontent.com/5444806/88010392-28de3d00-caeb-11ea-89a1-736cc261b5ba.png)

**Comment by rickybanister** [comment]
If we remove the `br` from the language code I think it looks just fine and has plenty of breathing room. 

Removing `font-weight; 600;` as well will help it align. There's no reason this should look different from other dropdowns. 

<img width="250" alt="Screen Shot 2020-07-21 at 9 23 13 AM" src="https://user-images.githubusercontent.com/437258/88060215-eee55900-cb33-11ea-9181-b1ab79c64e8d.png">

**Comment by ncardeli** [comment]
Ok, let's get this done 👍
